tests/loggers/test_cometml_logger.py

```diff
@@ -1,15 +1,122 @@
 # Copyright 2022 MosaicML Composer authors
 # SPDX-License-Identifier: Apache-2.0
-
 import os
 import zipfile
+from collections import defaultdict
 from json import JSONDecoder
 from pathlib import Path
 
 import pytest
+from torch.utils.data import DataLoader
+
+from composer.trainer import Trainer
+from tests.common import RandomClassificationDataset, SimpleModel
+
+
+def test_comet_ml_logging_train_loop(monkeypatch, tmp_path):
+    pytest.importorskip('comet_ml', reason='comet_ml is optional')
+    import comet_ml
+
+    monkeypatch.setattr(comet_ml, 'Experiment', comet_ml.OfflineExperiment)
+    from composer.loggers import CometMLLogger
+
+    # Set offline directory.
+    offline_directory = str(tmp_path / Path('.my_cometml_runs'))
+    os.environ['COMET_OFFLINE_DIRECTORY'] = offline_directory
+
+    comet_logger = CometMLLogger()
+
+    trainer = Trainer(
+        model=SimpleModel(),
+        train_dataloader=DataLoader(RandomClassificationDataset()),
+        train_subset_num_batches=2,
+        max_duration='2ep',
+        loggers=comet_logger,
+    )
+    trainer.fit()
+
+    del trainer
+
+    assert comet_logger.experiment is not None
+
```

Comment:
could you check for experiment ended here?
```suggestion
assert comet_logger.experiment.ended
```

Reply:
ah, this none check was to satisfy the type checker, but yes, i should add a check for ended as well